core/src/components/input/input.tsx

```diff
@@ -836,6 +857,30 @@ export class Input implements ComponentInterface {
                 <ion-icon aria-hidden="true" icon={mode === 'ios' ? closeCircle : closeSharp}></ion-icon>
               </button>
             )}
+            {this.showPasswordToggle && !readonly && !disabled && (
+              <button
+                aria-label="show password"
+                aria-checked={this.type === 'text' ? 'true' : 'false'}
```

Comment:
Since the implementation is accessing `type` multiple times in the function, it would be more performant to deconstruct `type` above and reference that variable reference. 

Accessing the variable from `this` will look up that reference in memory each time you access the property from it. 

For example:

```tsx
const { type } = this;

return (
  <button aria-checked={type === 'text' ? 'true' : 'false'}></button>
);
```

Reply:
Done!